core/client/src/client.rs

```diff
@@ -1180,6 +1180,50 @@ impl<B, E, Block, RA> Client<B, E, Block, RA> where
 		Ok(None)
 	}
 
+	fn get_descendants(&self, target: Block::Hash, maybe_skip: Option<Block::Hash>) -> error::Result<Vec<Block::Hash>> {
+		let mut children = self.backend.blockchain().children(target)?;
+		if let Some(skip) = maybe_skip {
+			children.retain(|&c| c != skip);
+		}
+		let mut descendants = Vec::new();
+		for child in children {
+			descendants.push(child);
+			let d = self.get_descendants(child, None)?;
+			descendants.extend(d);
+		}
+		Ok(descendants)
+	}
+
+	/// Gets the uncles of the block with `target_hash` going back `max_generation` ancestors.
+	pub fn uncles(&self, target_hash: Block::Hash, max_generation: NumberFor<Block>)
+		-> error::Result<Option<Vec<Block::Hash>>>
+	{
+		let load_header = |id: Block::Hash| {
+			match self.backend.blockchain().header(BlockId::Hash(id)) {
```

Comment:
```suggestion
			match self.backend.blockchain().header(BlockId::Hash(id))? {
```

And the you just need to match for `Some(_)` and `None`.